### PR TITLE
feat: add smali

### PR DESCRIPTION
**Pull request by ObserverOfTime**
I forked the grammar to fix an issue but I don't plan on maintaining the fork.
If the author ever merges my patch, I will switch to the original repository.

### CODE REVIEW

**Comment by clason** [comment]
What's the upstream then?

**Comment by ObserverOfTime** [comment]
https://git.sr.ht/~yotam/tree-sitter-smali

**Comment by clason** [comment]
And is the issue so bad that we can't use the original? Many parsers have bugs and pending PRs, and yet we use them. (Switching forks is not painless, so I prefer not to do it if it's not permanent and a significant improvement.)

**Comment by ObserverOfTime** [comment]
The original doesn't support character literals.

Guess I'll just remind the author to merge my patch.

**Comment by clason** [comment]
That would be preferable, yes. Let's see what happens.

File: queries/smali/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+(array_type
+  "[" @punctuation.bracket)
+
+; Methods
+
+(method_declaration
+  (method_identifier) @function)
+
+((statement) @function.call
+ (#lua-match? @function.call "^invoke"))
```
Comment (theHamsta) [inline queries/smali/highlights.scm:16]: isn't `invoke` a keyword? Could be `@keyword.function` although `@keyword.function` was only used until now for keywords introducing a function definition. `@keyword` would probably more general

Comment (ObserverOfTime) [reply queries/smali/highlights.scm:16]: This doesn't apply to `invoke-xyz` but to the function it calls.
I guess I could specifically target the function node though.

**Comment by theHamsta** [comment]
@ObserverOfTime is this still a draft? Are you planning to continue working on this?

**Comment by ObserverOfTime** [comment]
I'm waiting in case the author decides to merge my patch.

**Comment by amaanq** [comment]
The author's last commit was 11mo ago...so it could be a while if not never. I'd love to be able to have smali support as soon as I can :)

**Comment by theHamsta** [comment]
@amaanq do you want to open a new PR? I will close this one then you can create a new one.

**Comment by amaanq** [comment]
> @amaanq do you want to open a new PR? I will close this one then you can create a new one. Do you plan to maintain a fork?

Sure, I'd like to maintain a fork of the upstream repo actually...although I am not very familiar with making treesitter parsers and scheme. Hopefully it isn't too bad 😆 I'll open a new PR when ready.

**Pull request closed without merging**